core/ustring.cpp

```diff
@@ -586,14 +586,35 @@ String String::camelcase_to_underscore(bool lowercase) const {
 		bool is_upper = cstr[i] >= A && cstr[i] <= Z;
 		bool is_number = cstr[i] >= '0' && cstr[i] <= '9';
 		bool are_next_2_lower = false;
+		bool is_next_lower = false;
+		bool is_next_number = false;
 		bool was_precedent_upper = cstr[i - 1] >= A && cstr[i - 1] <= Z;
 		bool was_precedent_number = cstr[i - 1] >= '0' && cstr[i - 1] <= '9';
 
 		if (i + 2 < this->size()) {
 			are_next_2_lower = cstr[i + 1] >= a && cstr[i + 1] <= z && cstr[i + 2] >= a && cstr[i + 2] <= z;
 		}
 
-		bool should_split = ((is_upper && !was_precedent_upper && !was_precedent_number) || (was_precedent_upper && is_upper && are_next_2_lower) || (is_number && !was_precedent_number));
+		if (i + 1 < this->size()) {
+			is_next_lower = cstr[i + 1] >= a && cstr[i + 1] <= z;
+		}
+
+		if (i + 1 < this->size()) {
+			is_next_number = cstr[i + 1] >= '0' && cstr[i + 1] <= '9';
+		}
+
+		const bool a = is_upper && !was_precedent_upper && !was_precedent_number;
```

Comment:
I'd personally avoid variables named `a`, `b` and `c`, but their contents are fairly easy to read. This is a matter of taste, I don't know what's best @akien-mga ?.

Reply:
I'm not sure what a, b, and c do, but it was making the if statement massive.

```
const bool a = is_upper && !was_precedent_upper && !was_precedent_number;
const bool b = was_precedent_upper && is_upper && are_next_2_lower;
const bool c = is_number && !was_precedent_number && is_next_lower;
```

Have better names for them?

Reply:
I don't really have better ideas. I agree that the statement gets immense with bigger names. Let's wait for @akien-mga or any other contributor's opinion on this. As I said, these variables are initialized close from their usage, so it is fairly understandable.
If they were used elsewhere, I'd have no doubt on the need of finding a better name.